src/webrtc/groupCall.ts

```diff
@@ -495,12 +497,16 @@ export class GroupCall extends TypedEventEmitter<
             this.retryCallLoopInterval = undefined;
         }
 
+        if (this.participantsExpirationTimer !== null) {
+            clearTimeout(this.participantsExpirationTimer);
+            this.participantsExpirationTimer = null;
+        }
+
         if (this.state !== GroupCallState.Entered) {
             return;
         }
 
-        this.forEachCall((call) => this.disposeCall(call, CallErrorCode.UserHangup));
-        this.calls.clear();
```

Comment:
The calls map do not need clean up anymore?

Reply:
They should each be removed when the individual call is cleaned up. This was essentially doing it in a brute force way, but more importantly it was preempting the calls cleaning themselves up.